cura/Settings/MachineManager.py

```diff
@@ -474,6 +474,12 @@ def activeMachineId(self) -> str:
     def activeMachine(self) -> "GlobalStack":
         return self._global_container_stack
 
+    @pyqtProperty(str, notify = globalContainerChanged)
+    def activeMachineDefinitionId(self) -> str:
+        if self._global_container_stack and self._global_container_stack.definition:
+            return self._global_container_stack.definition.getId()
+        return ""
```

Comment:
The error is actually because we specified an incorrect type for the activeMachine property, it should be "QObject*" instead of "QObject".